crates/compiler/gen_llvm/src/llvm/expect.rs

```diff
@@ -219,7 +219,9 @@ fn build_clone<'a, 'ctx, 'env>(
             when_recursive,
         ),
 
-        Layout::LambdaSet(_) => unreachable!("cannot compare closures"),
+        // Since we will never actually display functions (and hence lambda sets), we can just pass
+        // back the nullptr.
+        Layout::LambdaSet(_) => env.ptr_int().const_zero(),
```

Comment:
@folkertdev  FYI. I think this is fine as per the comment, but let me know if you think this will be problematic.